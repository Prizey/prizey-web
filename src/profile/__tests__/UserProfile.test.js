import React from 'react'
import renderer from 'react-test-renderer'
import UserProfileInfo from '../UserProfile'

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New -
      <div>
        render -
        {props.render({
          create: jest.fn(),
          creating: false,
          error: null,
        })}
      </div>
      <div>
        renderCreated - {props.renderCreated && props.renderCreated({ id: 1 })}
      </div>
    </div>
  ),
}))

jest.mock('formik', () => ({
  Field: props => <div {...props}>Field</div>,
  Form: ({ children, ...props }) => <div {...props}>Form - {children}</div>,
  Formik: props => (
    <div {...props}>Formik - {props.render({ foo: 'bar' })}</div>
  ),
}))

jest.mock('design/Snackbar/Snackbar', () => props => (
  <div name="snackbar" {...props}>
    Snackbar
  </div>
))

it('renders correctly', () => {
  const params = {
    currentUser: true,
    setCurrentUser: jest.fn(),
  }
  const tree = renderer.create(<UserProfileInfo {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const params = {
    currentUser: false,
  }
  const tree = renderer.create(<UserProfileInfo {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})
