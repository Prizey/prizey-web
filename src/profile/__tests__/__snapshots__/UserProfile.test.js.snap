// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  afterCreate={[Function]}
  method="PUT"
  name="users"
  path="/auth"
  render={[Function]}
  renderCreated={[Function]}
>
  New -
  <div>
    render -
    <div
      className="Component-root-8"
    >
      <div
        className="Component-header-6"
      >
        <a
          aria-current="page"
          aria-label="Go Back"
          className="MuiButtonBase-root-16 MuiIconButton-root-10 Component-root-9"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiIconButton-label-15"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-19 MuiSvgIcon-fontSizeLarge-27"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-91"
          />
        </a>
        <div
          currentUser={true}
          setCurrentUser={
            [MockFunction] {
              "calls": Array [
                Array [
                  undefined,
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        >
          Logout
        </div>
      </div>
      <div
        className="Component-container-3"
      >
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <img
            alt="Prizey"
            className="Component-logoImage-7"
            src="/logo.png"
          />
        </a>
        <div
          className="Component-caption-1"
        />
        <div
          className="Component-children-2"
        >
          <h5
            className="MuiTypography-root-28 MuiTypography-h5-44 MuiTypography-alignCenter-51"
            style={
              Object {
                "marginTop": "-20px",
              }
            }
          >
            Profile
          </h5>
          <div
            initialValues={true}
            onSubmit={[MockFunction]}
            render={[Function]}
          >
            Formik - 
            <div
              className="Component-root-64"
            >
              Form - 
              <div
                component={[Function]}
                label="Full Name"
                name="fullname"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Address"
                name="address"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="City"
                name="city"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="State / Province / Region"
                name="stateProvinceRegion"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Postal Code / ZIP"
                name="zipcode"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Clothing Size"
                name="clothingSize"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Shoe Size"
                name="shoeSize"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="New Password"
                name="password"
                type="password"
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Password Confirmation"
                name="password_confirmation"
                type="password"
              >
                Field
              </div>
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              />
              <button
                className="MuiButtonBase-root-16 MuiButton-root-65 MuiButton-contained-76 MuiButton-containedPrimary-77 MuiButton-raised-79 MuiButton-raisedPrimary-80 MuiButton-fullWidth-90"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="submit"
              >
                <span
                  className="MuiButton-label-66"
                >
                  UPDATE PROFILE
                </span>
                <span
                  className="MuiTouchRipple-root-91"
                />
              </button>
            </div>
          </div>
          <a
            className="MuiButtonBase-root-16 MuiButton-root-65 MuiButton-outlined-73 MuiButton-outlinedPrimary-74 MuiButton-fullWidth-90"
            href="mailto:contact@prizey.app"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-66"
            >
              CONTACT US
            </span>
            <span
              className="MuiTouchRipple-root-91"
            />
          </a>
        </div>
        <div
          className="Component-footer-4"
        >
          <a
            className="Component-footerLink-5"
            href="/terms_of_service"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="/faq"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              FAIRNESS / FAQ
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="/all-prizes"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
    afterCreate - 
  </div>
</div>
`;
